src/wallet.cpp

```diff
@@ -1562,10 +1550,14 @@ bool CWallet::SelectCoinsForStaking(unsigned int nSpendTime,
             if (fDebug2 && fMiner)
                 LogPrintf("SelectCoinsForStaking: UTXO=%s (BalanceToConsider=%.8f >= Value=%.8f)", pcoin->vout[i].GetHash().ToString(), BalanceToConsider / (double)COIN, n / (double)COIN);
 
-            setCoinsRet.insert(make_pair(pcoin, i));
+            vCoinsRet.push_back(make_pair(pcoin, i));
         }
      }
 
+    // Randomize the vector order to keep PoS truely a roll of dice in which utxo has a chance to stake first
+    if (fMiner)
+        std::random_shuffle(vCoinsRet.begin(), vCoinsRet.end(), GetRandInt);
```

Comment:
Paul, I think std::random_shuffle is deprecated in C++14, because it depends on std::rand, which is also being deprecated... it will work now but may break soonish... as an alternative something like....

unsigned int seed = static_cast<unsigned int> GetAdjustedTime();

std::shuffle(vCoinsRet.begin(), vCoinsRet.end(), std::default_random_engine(seed));

Reply:
its used with the smallest utxos code. i'll change this and change the other location as well in a seperate pr. i didnt know we were going to c++ 14 anytime soon so didnt think about deprecation. 
also good point you pointed that out I looked at GetRandInt and i'll change the both locations to use random seeding.